libs/blog/articles/data-access/src/lib/state/article-details.store.ts

```diff
@@ -28,6 +29,7 @@ export const ArticleDetailsStore = signalStore(
   withCallState('fetch article details'),
   withMethods(({ ...store }) => {
     const articlesService = inject(ArticlesService);
+    const seoService = inject(SeoService);
```

Comment:
instead of this can you create signal store feature?
I'd like this to be hidden behind an abstraction, in case we would extend the seo.
```ts
export function withSeo() {
  return signalStoreFeature(
    ...
  );
}
``` 
^ and it would only have `withMethods`.